### ISSUE TITLE
Tabs navigation inconsistency with Code

### ISSUE THREAD

**Issue by mahmoudajawad** [eclipse-theia/theia#9036]
> I'm considering this not to be a bug, but a behaviour difference with Code, in which I'm falling for categorising it as a feature request.

### Feature Description:
Navigating between opened tabs in Code uses the same key-binding; Ctrl+tab, Ctrl+Shift+tab. However, the behaviour is different; In Theia the key-binding selects the next and the previous opened tabs, while in Code it selects the last focused and the next-to-last focused tabs, in order.

I tried finding any shortcut set that could replicate the behaviour of Code in Theia but couldn't, and it seems to be requiring built-in support.

**Comment by vince-fugnitto** [comment]
@mahmoudajawad thank you for the issue, I believe the keybindings are different in order not to clash with the browser keybindings (which switches browser tabs). 

- <kbd>ctrl</kbd>+<kbd>tab</kbd>
- <kbd>ctrl</kbd>+<kbd>shift</kbd>+<kbd>tab</kbd>

These are only the default keybindings that the framework provides, if you wish to update them for your own product you may do so by customizing them through a custom extension, else as a user you may update them in your `keymaps.json`. Please be aware however of the limitation on the browser.

**Comment by mahmoudajawad** [comment]
@vince-fugnitto, oops! I did forget that I changed the default tab switching key-bindings from:
```
View: Switch to Next Tab | ctrlcmd+alt+d
View: Switch to Previous Tab | ctrlcmd+alt+a
```
to:
```
View: Switch to Next Tab | ctrl+tab
View: Switch to Previous Tab | ctrl+shift+tab
```
as I'm using Thiea in chromium app mode. Though, this isn't my concern--Irregardless of the key-bindings in use, the behaviour itself differs from code. So, while code goes to the last focused tab, Theia goes to the next opened tab in the order. The behaviour can only be noticed if you have more than two tabs opened and not only two.

**Comment by vince-fugnitto** [comment]
@mahmoudajawad I believe we opted for such a simple implementation since we do not support the `editorPicker` from which the keybindings trigger:
- `workbench.action.quickOpenNavigateNextInEditorPicker`
- `workbench.action.quickOpenNavigatePreviousInEditorPicker`

The following is the behavior I noticed when triggering <kbd>ctrl</kbd>+<kbd>tab</kbd> (linux):

![tab-picker](https://user-images.githubusercontent.com/40359487/107368203-40dcd480-6aae-11eb-949d-7015b73dff27.gif)

**Comment by mahmoudajawad** [comment]
Yup, this is the behaviour that I'm after in Theia. I'm not questioning the back-story of why this isn't the behaviour, as I understand that Theia made what is considered the better decisions for it, I'm just requesting this as an alternative to what is currently implemented in Theia, and possibly keep both the options.

**Comment by vince-fugnitto** [comment]
>  I'm just requesting this as an alternative to what is currently implemented in Theia, and possibly keep both the options.

I think it's worthwhile to introduce the functionality of the `editorPicker` (we should probably open a corresponding issue to track it and close this one). If interested, please feel free to provide a pull-request.

**Comment by mahmoudajawad** [comment]
I definitely would love to contribute this feature to Theia. I'll look into how I can do so, and hopefully I can find sometime to actually work on it soon.

**Comment by vince-fugnitto** [comment]
@airdrik thank you for the response, it has been a while since the original author posted so we thought the feature wasn't a high priority. Do you mind sharing what you expect as a behavior, are you looking for the feature in the desktop or browser use-case? (note: in the browser the <kbd>ctrl</kbd>+<kbd>tab</kbd> is reserved so it would not work, similarly to how I believe it doesn't for vscode in the web).

Note that we have the commands `View: Show Next Tab` and `View: Show Previous Tab` which you can assign keybindings to that work in the browser to cycle the tabs.

**Comment by airdrik** [comment]
I use theia-blueprint as a desktop/electron app.  I've configured `View: Show Next Tab` and `View: Show Previous Tab` to use `ctrl+tab` and `ctrl+shift+tab`, respectively. Currently when repeating the `ctrl+tab` sequence, it always switches to the next tab to the right of the current tab, while repeating `ctrl+shift+tab` always switches to the next tab to the left of the current tab.  I've also switched to using the `workbench.action.nextEditor` and `workbench.action.previousEditor` actions, but those provide basically the same behavior as `View: Show Next Tab` and `View: Show Previous Tab`. 

The MRU switching behavior I'd like to see implemented would work as follows:
Hitting `ctrl+tab` (`View: Show Next Tab`, or perhaps a new action: show last used tab) will switch to the last tab you were on previously, then hitting `ctrl+tab` again would bring you back to the same tab you started on (because that is now the most-recently-used tab).  Holding `ctrl` and repeatedly hitting `tab` would cycle through the tabs in order by when they were last used (possibly accompanied by a widget/drop-down showing the open tabs in their MRU order).  The next tab is only actually selected when lifting `ctrl` (such that pressing `ctrl+tab` again would bring you back to the tab you were on when you initiated the switching).
Hitting `ctrl+shift+tab` (`View: Show Previous Tab` or a new action: tab switch - reverse) would bring you to the least-recently-used tab, while holding `ctrl+shift` and repeatedly hitting `tab` would cycle through the tabs in reverse order (and similarly show the widget/drop-down with the open tabs in their MRU order).  Holding and releasing `shift` while cycling through the tabs would toggle whether pressing `tab` will move forward or backward in the list.

For example, say I have 4 tabs open, with files A, B, C and D, which I had opened in that order (so the MRU ordering would be D, C, B, A) and am currently looking at D:
Say I hold `ctrl` and hit `tab`; then the list of tabs should appear with tab C selected
If I then press and hold `shift`, followed by hitting `tab`, then it should select the next-recently-used tab (tab D, back to where I currently am).
If I then release `shift` and then hit `tab` twice (selecting tab B) and release `ctrl`, the widget should disappear and the editor should switch to tab B.
The MRU ordering would now be B, D, C, A. If I then were to hit (and release) `ctrl+tab`, then it would switch back to tab D (and the MRU ordering would be D, B, C, A); etc.

I would expect to see an option in Preferences to select which behavior (the current behavior vs. this MRU behavior) is used for switching tabs.
